src/core/ddsi/src/q_init.c

```diff
@@ -1403,7 +1403,8 @@ int rtps_start (struct q_globals *gv)
   }
   if (gv->listener)
   {
-    (void) create_thread (&gv->listen_ts, gv, "listen", (uint32_t (*) (void *)) listen_thread, gv->listener);
+    if (create_thread (&gv->listen_ts, gv, "listen", (uint32_t (*) (void *)) listen_thread, gv->listener) != DDS_RETCODE_OK)
+      GVERROR ("rtps_start: can't create listener thread\n");
```

Comment:
Stopping everything is quite intricate (which no doubt is the reason that the error handling was omitted in the first place), and logging that something went badly wrong is a definite improvement over silence. But if you look at the start/stop sequence, ``rtps_stop`` undoes ``rtps_start`` and not much is needed to use ``rtps_stop`` for cleaning up in these two error cases. Looking at ``rtps_stop`` you'd think:

```C
ddsi_listener_free(gv->listener);
gv->listener=NULL;
rtps_stop();
```

would do the trick. That's far more elegant than simply printing an error.